File: backend/src/main/java/es/codeurjc/webapp14/dto/ReviewReportedDTO.java
package es.codeurjc.webapp14.dto;

import java.util.List;

public record ReviewReportedDTO(
    Long id,
    String username,
    ProductDTO product,
    int rating,
    String reviewText,
    boolean reported,
    boolean own,
    List<Boolean> ratingStars,
    List<Boolean> emptyStars
) {}

File: frontend/src/app/dtos/reviewReported.dto.ts
import { ProductDTO } from "./product.dto";
import { UserDTO } from "./user.dto";


export interface ReviewReportedDTO {
    id?: number,
    username: string,
    user: UserDTO,
    product: ProductDTO,
    rating: number,
    reviewText: String,
    reported: boolean,
    own: boolean,
    ratingStars: boolean[],
    emptyStars: boolean[]
}
